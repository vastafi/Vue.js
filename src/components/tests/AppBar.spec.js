import {createLocalVue,shallowMount} from "@vue/test-utils";
import AppBar from "../AppBar"
import Vuex from 'vuex'
import Vuetify from 'vuetify';
import VueRouter from "vue-router";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);
localVue.use(VueRouter);
const createElement = (element) => localVue.component(element,{
    functional:true,
    // eslint-disable-next-line no-unused-vars
    function(createElement, hack){
        return createElement('div')
    }
})

localVue.component('VIcon',createElement("VIcon"));
localVue.component('v-app-bar',{})
localVue.component('v-container',{})
localVue.component('v-toolbar-items',{})

const store = new Vuex.Store({
    getters:{
        'products/getSearchSuggestions': jest.fn(),
        'products/getIsSearchLoading': jest.fn(),
        'settings/getIsDarkModeEnabled':jest.fn()
    }
})
//
// describe('AppBar', () => {
//     it('should match snapshot',()=> {
//         const component = mount(AppBar,{
//             store,
//             localVue,
//             vuetify: new Vuetify({
//                 icons:{
//                     iconfont:'fa'
//                 }
//             })
//         });
//         expect(component.element).toMatchSnapshot();
//     })
// })

describe('AppBar', () => {
    it('should match snapshot',()=> {
        const component = shallowMount(AppBar,{
            store,
            localVue,
            vuetify: new Vuetify({
                icons:{
                    iconfont:'fa'
                }
            })
        });
        expect(component.element).toMatchSnapshot();
    })
})