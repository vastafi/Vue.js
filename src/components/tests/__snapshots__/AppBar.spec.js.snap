// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar should match snapshot 1`] = `
<div>
  <v-app-bar-stub
    app=""
    class="indigo"
    flat=""
  >
    <v-container-stub
      class="py-0 fill-height"
    >
      <v-toolbar-items-stub
        class="hidden-lg-and-up"
      >
        <v-app-bar-nav-icon-stub />
      </v-toolbar-items-stub>
       
      <v-btn-stub
        activeclass=""
        class="hidden-md-and-down"
        color="black"
        tag="button"
        text="true"
        to="/products"
        type="button"
      >
        
        Products
      
      </v-btn-stub>
      <v-btn-stub
        activeclass=""
        class="hidden-md-and-down"
        color="black"
        tag="button"
        text="true"
        to="/contacts"
        type="button"
      >
        
        Contacts
      
      </v-btn-stub>
      <v-btn-stub
        activeclass=""
        class="hidden-md-and-down"
        color="black"
        tag="button"
        text="true"
        to="/about"
        type="button"
      >
        
        About
      
      </v-btn-stub>
       
      <v-spacer-stub />
       
      <search-stub
        id="search"
        items=""
        value=""
      />
       
      <v-spacer-stub />
       
      <div
        class="authorization"
      >
        <router-link-stub
          ariacurrentvalue="page"
          class="auth-links"
          event="click"
          id="log-in"
          tag="a"
          to="[object Object]"
        >
          <v-icon-stub>
            mdi-login-variant
          </v-icon-stub>
           
          <div
            class="auth-title hidden-sm-and-down"
          >
            
            Sign In
          
          </div>
        </router-link-stub>
         
        <router-link-stub
          ariacurrentvalue="page"
          class="auth-links"
          event="click"
          id="sign-up"
          tag="a"
          to="[object Object]"
        >
          <v-icon-stub>
            mdi-account-plus
          </v-icon-stub>
           
          <div
            class="auth-title hidden-sm-and-down"
          >
            
            Sign Up
          
          </div>
        </router-link-stub>
      </div>
    </v-container-stub>
  </v-app-bar-stub>
   
  <v-navigation-drawer-stub
    fixed="true"
    height="100%"
    minivariantwidth="56"
    mobilebreakpoint="1264"
    src=""
    tag="aside"
    temporary="true"
    width="256"
  >
    <categories-stub
      class="col-md-12 col-12"
    />
  </v-navigation-drawer-stub>
</div>
`;
